shareMethod.sh

```diff
@@ -25,7 +25,16 @@ execWithTimeout() {
     i=1
     while [[ $i -le $retry ]]
     do
-        expect -c "set timeout $timeout; spawn -noecho $cmd; expect timeout { exit 1 } eof { exit 0 }"
+        expect << 'EOF'
+            set timeout $env(timeout)
+            eval spawn -noecho $env(cmd)
+            expect {
+                timeout { exit 1 }
+                eof
+            }
+            lassign [wait] pid spawnid os_error_flag value
```

Comment:
`lassign  [wait] pid spawnid os_error_flag value` can get the exit code of spawned process in expect shell script. And the original `expect -c "set timeout $timeout; spawn -noecho $cmd; expect timeout { exit 1 } eof { exit 0 } ` can not, which always returns 0 when `spawn -noecho $cmd` failed.